RetailCoder.VBE/Root/RubberduckIoCInstaller.cs

```diff
@@ -174,6 +177,13 @@ private void RegisterConfiguration(IWindsorContainer container, Assembly[] assem
                 .ImplementedBy<SourceControlConfigProvider>()
                 .LifestyleSingleton());
 
+            container.Register(Component.For<IConfigProvider<HotkeySettings>>()
+                .ImplementedBy<HotkeyConfigProvider>()
+                .DynamicParameters((kernel, parameters) => parameters["defaultHotkeys"] = kernel.Resolve<IEnumerable<CommandBase>>()
+                        .Where(command => command.DefaultHotkey != null)
+                        .Select(command => command.DefaultHotkey))
```

Comment:
Why should the procedure to identify default hotkeys be a responsibility of the IoC registration? How about introducing a `DefaultHotkeys` class taking an `IEnumerable<CommandBase>` as constructor parameter and having a single property `DefaultHotkeys`? Then, everything needing the default settings could just take an instance of this class, which should be registered in singleton scope to self in the IoC container.

Reply:
Makes perfect sense. The thing is injecting this class into `HotkeyConfigProvider` will result in a circular dependency. Maybe @Vogel612's comments will help me redesign things better.